### PR TITLE
Azure ML basic workflow

### PR DESCRIPTION
**Pull request by EwanNoble**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -23,14 +21,24 @@ resources:
   - repository: das-platform-building-blocks
     type: github
     name: SkillsFundingAgency/das-platform-building-blocks
-    ref: refs/tags/0.3.31
+    ref: refs/heads/ml
```
Comment (NickGraham101) [inline azure-pipelines.yml:24]: Should ref a tag

File: azure/template.json [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+                "templateLink": {
+                    "uri": "[concat(variables('deploymentUrlBase'),'machine-learning-services/machine-learning-datastore-blob.json')]",
+                    "contentVersion": "1.0.0.0"
+                },
+                "parameters": {
+                    "workspaceName": {
+                        "value": "[variables('workspaceName')]"
+                    },
+                    "datastoreName": {
+                        "value": "trainingdata"
```
Comment (NickGraham101) [inline azure/template.json:337]: Could use `[variables('dataStorageContainerName')]` here if we're going to use the same name for the Storage Account Blob Container and the ML Data Store

File: pipeline-templates/job/build-and-train.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+parameters:
+  Environment:
+  ServiceConnection:
+
+jobs:
+  - deployment: DeployInfra
```
Comment (NickGraham101) [inline pipeline-templates/job/build-and-train.yml:6]: Can the list indenting be removed?  Convention is to align list items in the same column as the parent property

File: models/placeholder/train.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import pickle
+import os
+from azureml.core.run import Run
+from azureml.core import Datastore
+
```
Comment (NickGraham101) [inline models/placeholder/train.py:5]: Remove extra whitespace line

File: pipeline-templates/job/build-and-train.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        value: $[ dependencies.DeployInfra.outputs['DeployInfra.ArmOutputs${{ parameters.Environment }}.WorkspaceName'] ]
+      - name: ResourceGroupName
+        value: $[ dependencies.DeployInfra.outputs['DeployInfra.ArmOutputs${{ parameters.Environment }}.ResourceGroupName'] ]
+      - name: ComputeName
+        value: $[ dependencies.DeployInfra.outputs['DeployInfra.ArmOutputs${{ parameters.Environment }}.ComputeName'] ]
+    steps:
+      - checkout: self
+      - task: AzureCLI@2
+        displayName: "Train and Register"
+        inputs:
```
Comment (NickGraham101) [inline pipeline-templates/job/build-and-train.yml:35]: This should be refactored but on the assumption we'll be returning to this pipeline that can (and probably should) wait till it's been used a bit more by the team

**Review by NickGraham101 — CHANGES_REQUESTED** [review]

File: azure/template.json [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+                    }
+                }
+            },
+            "dependsOn": [
+                "[variables('resourceGroupName')]"
+            ]
+        },
+        {
+            "apiVersion": "2020-10-01",
+            "name": "ml-default-storage",
```
[the remark this answers is not in the document]
Comment (EwanNoble) [reply azure/template.json:119]: ML Studio seemed to refuse to access it's default/attached storage account via AAD which meant I was unable to disable shared key access and couldn't enable network ACLs. It just uses it to store logs and artifacts as far as I can tell so the storage account which will contain actual data will be separate and more secure.

File: pipeline-templates/job/build-and-train.yml (hunk 5 above)
Comment (EwanNoble) [reply pipeline-templates/job/build-and-train.yml:35]: Yes agreed, I don't think this is the long-term solution but needs to get some use to see where the issues are

**Review by NickGraham101 — APPROVED** [review]

**Pull request merged**